program/tests/helpers/wrap_builder.rs

```diff
@@ -0,0 +1,299 @@
+    unwrapped_escrow_account: Option<Account>,
+    unwrapped_token_program: Option<TokenProgram>,
+    wrapped_token_program: Option<TokenProgram>,
+}
+
+impl<'a> Default for WrapBuilder<'a> {
+    fn default() -> Self {
+        Self {
+            mollusk: init_mollusk(),
+            wrap_amount: None,
+            recipient: None,
+            checks: vec![],
+            wrapped_mint: None,
+            unwrapped_escrow_addr: None,
+            wrapped_mint_authority: None,
+            unwrapped_escrow_owner: None,
+            recipient_starting_amount: None,
+            unwrapped_token_starting_amount: None,
+            unwrapped_escrow_account: None,
+            unwrapped_token_program: None,
+            wrapped_token_program: None,
+        }
+    }
+}
+
+impl<'a> WrapBuilder<'a> {
+    pub fn wrap_amount(mut self, amount: u64) -> Self {
+        self.wrap_amount = Some(amount);
+        self
+    }
+
+    pub fn unwrapped_token_starting_amount(mut self, amount: u64) -> Self {
+        self.unwrapped_token_starting_amount = Some(amount);
+        self
+    }
+
+    pub fn wrapped_mint(mut self, account: KeyedAccount) -> Self {
+        self.wrapped_mint = Some(account);
+        self
+    }
+
+    pub fn wrapped_token_program(mut self, program: TokenProgram) -> Self {
+        self.wrapped_token_program = Some(program);
+        self
+    }
+
+    pub fn unwrapped_token_program(mut self, program: TokenProgram) -> Self {
+        self.unwrapped_token_program = Some(program);
+        self
+    }
+
+    pub fn unwrapped_escrow_addr(mut self, key: Pubkey) -> Self {
+        self.unwrapped_escrow_addr = Some(key);
+        self
+    }
+
+    pub fn unwrapped_escrow_account(mut self, account: Account) -> Self {
+        self.unwrapped_escrow_account = Some(account);
+        self
+    }
+
+    pub fn unwrapped_escrow_owner(mut self, key: Pubkey) -> Self {
+        self.unwrapped_escrow_owner = Some(key);
+        self
+    }
+
+    pub fn wrapped_mint_authority(mut self, key: Pubkey) -> Self {
+        self.wrapped_mint_authority = Some(key);
+        self
+    }
+
+    pub fn recipient_starting_amount(mut self, amount: u64) -> Self {
+        self.recipient_starting_amount = Some(amount);
+        self
+    }
+
+    pub fn check(mut self, check: Check<'a>) -> Self {
+        self.checks.push(check);
+        self
+    }
+
+    fn get_wrapped_mint(
+        &self,
+        token_program: TokenProgram,
+        unwrapped_mint_addr: Pubkey,
+    ) -> KeyedAccount {
+        let wrapped_mint_addr = get_wrapped_mint_address(&unwrapped_mint_addr, &token_program.id());
+        let mint_authority = get_wrapped_mint_authority(&wrapped_mint_addr);
+
+        self.wrapped_mint.clone().unwrap_or(KeyedAccount {
+            key: wrapped_mint_addr,
+            account: setup_mint(token_program, &self.mollusk.sysvars.rent, mint_authority),
+        })
+    }
+
+    pub fn setup_token_account(&self, wrapped_mint: &KeyedAccount) -> KeyedAccount {
```

Comment:
Although the mint on token-2022 is correctly configured with the `TransferFeeConfig`, the token accounts need to also be configured with the `TransferFeeAmount` extension, otherwise the token accounts won't work properly. Also, we won't get to test the behavior of when a token account has an extension

Reply:
Ah I see. Updated this with proper extension initialization.